app/src/features/rules/screens/rulesList/components/RulesList/components/ModheaderImporter/ModheaderImporter.tsx

```diff
@@ -0,0 +1,239 @@
+
+    reader.onerror = () => {
+      setValidationError("Could not process the selected file! Try again.");
+      return;
+    };
+
+    reader.onload = () => {
+      const fileContent = reader.result;
+      setIsDataProcessing(true);
+
+      if (!file.type.includes("json")) {
+        // stop parsing for wrong file format
+        setIsDataProcessing(false);
+        setValidationError("Failed to parse Modheader file.");
+
+        return;
+      }
+
+      let modheaderProfilesJSON;
+      let parsedRules = [];
+      try {
+        modheaderProfilesJSON = JSON.parse(fileContent as string);
+        parsedRules = parseRulesFromModheader(modheaderProfilesJSON);
+      } catch (e) {
+        setValidationError("Failed to parse your Modheader file.");
+        setIsDataProcessing(false);
+        return;
+      }
+
+      setIsDataProcessing(false);
+      setRulesToImport(parsedRules);
+      setIsParseComplete(true);
+    };
+    reader.readAsText(file);
+  };
+
+  const handleModheaderRulesImport = () => {
+    setIsLoading(true);
+
+    addRulesAndGroupsToStorage(appMode, rulesToImport)
+      .then(() => {
+        callback?.();
+        navigate(PATHS.RULES.ABSOLUTE);
+      })
+      .catch((e) => {
+        setValidationError("Failed to import your Modheader file.");
+        return;
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  const handleResetImport = () => {
+    setValidationError(null);
+    setIsParseComplete(false);
+    setRulesToImport([]);
+  };
+
+  const isParsedRulesExist = useMemo(() => !!rulesToImport.length, [rulesToImport?.length]);
+
+  return (
+    <>
+      <div className="modheader-import-container">
+        <Row justify={"space-between"} className="modheader-import-header">
+          <Col className="modheader-import-heading">
+            {isBackButtonVisible && (
+              <ArrowLeftOutlined size={16} className="modheader-import-back-icon" onClick={onBackButtonClick} />
+            )}
+            Import Modheader profiles
+          </Col>
+          <Col
+            className="modheader-import-share-container"
+            onClick={() =>
+              copyToClipBoard(window.origin + PATHS.IMPORT_FROM_MODHEADER.ABSOLUTE, "URL copied to clipboard")
+            }
+          >
+            <LinkOutlined className="icon__wrapper" />
+            Share
+          </Col>
+        </Row>
+
+        {!isParseComplete && !validationError && (
+          <FilePicker
+            maxFiles={1}
+            onFilesDrop={(files) => {
+              handleResetImport();
+              onFilesDrop(files);
+            }}
+            isProcessing={isDataProcessing}
+            title="Drag and drop your Modheader export file to upload"
+            subtitle="Accepted file formats: json"
+            selectorButtonTitle={isParseComplete || validationError ? "Try another file" : "Select file"}
+          />
+        )}
+
+        {(isParseComplete || validationError) && (
```

Comment:
There is a same condition block at line 183. Both the markups can be accommodated in single `(isParseComplete || validationError)` block.